core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -270,18 +270,20 @@ private[spark] class CoarseGrainedExecutorBackend(
                              reason: String,
                              throwable: Throwable = null,
                              notifyDriver: Boolean = true) = {
-    val message = "Executor self-exiting due to : " + reason
-    if (throwable != null) {
-      logError(message, throwable)
-    } else {
-      logError(message)
-    }
+    if (stopping.compareAndSet(false, true)) {
```

Comment:
If stopping is true, we should add a log saying we entered the `exitExecutor` function, but skipped actual actions because we already tried to exit before.

Reply:
Added.